sync-core/src/test/java/com/cloudant/sync/query/QueryFilterFieldsTest.java

```diff
@@ -0,0 +1,203 @@
+import com.cloudant.sync.datastore.DocumentRevision;
+import com.cloudant.sync.datastore.MutableDocumentRevision;
+import com.cloudant.sync.datastore.ProjectedDocumentRevision;
+import com.cloudant.sync.util.SQLDatabaseTestUtils;
+
+import org.junit.Assert;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryFilterFieldsTest extends AbstractQueryTestSetUp {
+
+    @Override
+    public void setUp() throws SQLException {
+        super.setUp();
+        im = new IndexManager(ds);
+        assertThat(im, is(notNullValue()));
+        db = im.getDatabase();
+        assertThat(db, is(notNullValue()));
+        assertThat(im.getQueue(), is(notNullValue()));
+        String[] metadataTableList = new String[] { IndexManager.INDEX_METADATA_TABLE_NAME };
+        SQLDatabaseTestUtils.assertTablesExist(db, metadataTableList);
+
+        setUpBasicQueryData();
+    }
+
+    // When filtering fields on find
+
+    @Test
+    public void returnsFieldSpecifiedOnly() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query, 0, Long.MAX_VALUE, Arrays.asList("name"), null);
+        for (DocumentRevision rev : queryResult) {
+            Map<String, Object> revBody = rev.getBody().asMap();
+            assertThat(revBody.keySet(), contains("name"));
+        }
+    }
+
+    @Test
+    public void returnsAllFieldsWhenFieldsArrayEmpty() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query, 0, Long.MAX_VALUE, new ArrayList<String>(), null);
+        for (DocumentRevision rev : queryResult) {
+            Map<String, Object> revBody = rev.getBody().asMap();
+            assertThat(revBody.keySet(), containsInAnyOrder("name", "pet", "age"));
+        }
+    }
+
+    @Test
+    public void returnsAllFieldsWhenFieldsArrayNull() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query, 0, Long.MAX_VALUE, null, null);
+        for (DocumentRevision rev : queryResult) {
+            Map<String, Object> revBody = rev.getBody().asMap();
+            assertThat(revBody.keySet(), containsInAnyOrder("name", "pet", "age"));
+        }
+    }
+
+    @Test
+    public void returnsNullWhenUsingDottedNotation() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        assertThat(im.find(query, 0, Long.MAX_VALUE, Arrays.asList("name.blah"), null),
+                is(nullValue()));
```

Comment:
Fixed in https://github.com/cloudant/sync-android/commit/fde66732f755b51b30bbea43a00d30672e1fd403